File: example/jchatbot/src/main/java/zw/co/dcl/jchatbot/configs/SessionConfig.java
package zw.co.dcl.jchatbot.configs;


import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Component;
import zw.co.dcl.jawce.session.ISessionManager;
import zw.co.dcl.jawce.session.impl.CaffeineSessionManager;
import zw.co.dcl.jawce.session.impl.FileBasedSessionManager;

import java.util.concurrent.TimeUnit;


@Component
public class SessionConfig {
    @Bean
    public ISessionManager sessionManager() {
        // TODO: ISessionManager implement or use your own choice
        // return FileSessionManager.getInstance();
        return CaffeineSessionManager.getInstance(30L, TimeUnit.MINUTES);
    }
}

File: chatbot/src/main/kotlin/zw/co/dcl/jawce/chatbot/configs/SessionConfig.kt
package zw.co.dcl.jawce.chatbot.configs

import org.springframework.context.annotation.Bean
import org.springframework.stereotype.Component
import zw.co.dcl.jawce.session.ISessionManager
import zw.co.dcl.jawce.session.impl.FileBasedSessionManager

@Component
class SessionConfig {
    @Bean
    fun sessionManager(): ISessionManager {
        return FileBasedSessionManager.getInstance();
    }
}
